src/workerd/jsg/README.md

```diff
@@ -708,16 +708,24 @@ the C++ class to the JavaScript implementation of it such that the constructor (
 is automatically routed to the static `Foo::constructor()` method, and the getter accessor for
 the `value` property is automatically routed to the `Foo::getValue()` method.
 
-JSG intentionally hides away most of the details of how this works so we're not going to cover
-that piece here.
+JSG intentionally hides away most of the details of how this works, so we're not going to cover
+those here.
 
 The example illustrates all of the top-level pieces of defining a Resource Type but there are
 a range of details. We'll cover those next.
 
+// r: it feels like we're discussing jsg::Object only for use with Resource Types / as facilitating them,
+//    but don't they have other uses aside from that? It seems like jsg::Object should be discussed on
+//    its own, probably before introducing it as a vehicle for implementing Resource Types, because it
+//    contains a lot of the plumbing necessary for any kind of object that does a little more behind the
+//    scenes than just being a "dumb struct"...
+//    but maybe I'm just getting confused by the "Resource Type" moniker, and really I should just think
+//    of it as "smart object" (vs "dumb struct"), and then jsg::Object is of course exactly the right place
+//    to discuss it, because basically that's literally what "Resource Type" translates to in terms of C++
```

Comment:
`jsg::Object` is *only* used as the base class for resource types. They have no other use.